### PR TITLE
Relative date format feature

### PR DESCRIPTION
**Pull request by 123InktJules**
Format date relative to today.

### CODE REVIEW

File: src/DateFormatService.php [hunk 1]
```diff
@@ -4,48 +4,53 @@
...
 {
-    /** @var array<string, IntlDateFormatter> */
-    private array $formatters = [];
-
-    public function __construct(private string $locale, private string $timezone)
-    {
+    private DateFormatHelper $dateFormatHelper;
+    private RelativeDateFallbackService $fallbackHandler;
+
+    public function __construct(
```
Comment (frankdekker) [inline src/DateFormatService.php:17]: Please also update the readme in the root of the project with the new feature

Comment (123InktJules) [reply src/DateFormatService.php:17]: Readme updated

File: src/DateFormatService.php [hunk 2]
```diff
@@ -4,48 +4,53 @@
...
+        private readonly DateFormatOptions $options,
```
Comment (frankdekker) [inline src/DateFormatService.php:18]: Also add `private readonly ?RelativeDateFormatOptions  $relativeOptions = null`, to allow setting a global default option

File: src/DateFormatService.php [hunk 3]
```diff
@@ -4,48 +4,53 @@
...
     }
 
-    private function getDateFormatter(string $format): IntlDateFormatter
-    {
-        if (isset($this->formatters[$format])) {
-            return $this->formatters[$format];
+    public function formatRelative(
+        int|string|DateTimeInterface $value,
+        string                       $pattern,
+        RelativeDateFormatOptions    $relativeOptions,
```
Comment (frankdekker) [inline src/DateFormatService.php:42]: Make this optional and nullable.

File: README.md [hunk 4]
```diff
@@ -100,6 +102,29 @@ $dateFormatter->format(new DateTime(), 'eeee dd LLLL Y - HH:mm:ss');
...
+
+```php
+$dateFormatOptions = new DateFormatOptions('nl_NL', date_default_timezone_get())
+$dateFormatter = new DateFormatService($dateFormatOptions);
+
+$dateFormatter->formatRelative(time(), 'Y-m-d', new RelativeDateFormatOptions(1));
+// example output: Vandaag
+
+$dateFormatter->formatRelative(new DateTime('+1 day'), 'Y-m-d', new RelativeDateFormatOptions(1);
+// example output: Morgen
```
Comment (frankdekker) [inline README.md:116]: Is `Gisteren` also possible with this library? If so, add that as additional example

File: src/Date/RelativeDateFallbackResult.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DR\Internationalization\Date;
+
+class RelativeDateFallbackResult
+{
+    public function __construct(private readonly bool $fallback, private readonly string|bool $date = '')
```
[the remark this answers is not in the document]
Comment (frankdekker) [reply src/Date/RelativeDateFallbackResult.php:9]: It seems you get string|false back from IntlDateFormatter. I recommend we make this library strict and convert `false` to exception.

File: src/Date/DateFormatHelper.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public function getParsedDate(int|string|DateTimeInterface $date): DateTimeInterface
+    {
+        if (is_string($date)) {
+            return new DateTimeImmutable($date);
+        }
+
+        if (is_int($date)) {
+            return new DateTimeImmutable('@' . $date);
```
Comment (frankdekker) [inline src/Date/DateFormatHelper.php:36]: I think we should use `(new DateTimeImmutable())->setTimestamp($date)` here. 

Your version will always set the timezone to UTC. My version will honor the systems timezone.

**Review by frankdekker — APPROVED** [review]

**Pull request merged**